igvc_perception/include/pointcloud_filter/fast_segment_filter/fast_segment_filter.h

```diff
@@ -0,0 +1,101 @@
+#ifndef GROUNDSEGMENTER_H
+#define GROUNDSEGMENTER_H
+
+#include <math.h>
+#include <pcl/point_cloud.h>
+#include <pcl_ros/point_cloud.h>
+#include <pointcloud_filter/fast_segment_filter/fast_segment_filter_config.h>
+#include <pointcloud_filter/filter.h>
+#include <velodyne_pointcloud/point_types.h>
+#include <visualization_msgs/Marker.h>
+#include <visualization_msgs/MarkerArray.h>
+
+#include <Eigen/Dense>
+
+typedef Eigen::Matrix<double, Eigen::Dynamic, 3> MatrixXd;
+typedef Eigen::Matrix<double, Eigen::Dynamic, 1> VectorXd;
+
+namespace pointcloud_filter
+{
+struct Prototype
+{
+  double distance_;
+  velodyne_pointcloud::PointXYZIR point_;
+  double blue_;
+  double red_;
+  double green_;
+
+  bool operator<(const Prototype &p) const
+  {
+    return distance_ < p.distance_;
+  }
+};
```

Comment:
The purpose of this struct doesn't seem to be very obvious from its name... Maybe add a doxygen comment?